src/audio-worklet/migration/spn-recorder/app.js

```diff
@@ -101,6 +112,21 @@ function setupScriptProcessor(recordBuffer, passSampleToVisualizers) {
     // Update tracked recording length.
     if (isRecording) {
       recordingLength += BUFFER_SIZE;
+      
+      if (recordingLength > recordingProperties.maxFrameCount) {
+        isRecording = !isRecording;
+
+        const finalRecordBuffer =
+            createFinalRecordBuffer(recordingProperties);
+        const wavUrl = createLinkFromAudioBuffer(finalRecordBuffer, true);
+        player.src = wavUrl;
+        downloadLink.href = wavUrl;
+        downloadLink.download =
+            `recording-${new Date().getMilliseconds().toString()}.wav`;
+        downloadButton.disabled = false;
+        recordText.innerHTML = 'Ready to download 5 mins';
```

Comment:
`innerHTML` to `textContent`.

Also are we downloading a "5-minute file" always?

Reply:
No, the newest algorithm can make the download always be the right length.

Reply:
Then we need to change the text to reflect what's happening.